### PR TITLE
Fixing password confirmation

### PR DESCRIPTION
**Pull request by Lydias303**
This has the current master merged in and all tests are passing. 

SO the password confirmation wasnt working right and i had to change it to be validated on create. 

I had to add back in the current_user update in the user listings create method as it some how got removed.

I had to change all the test to have password confirmation in all their create methods.

this was originally 3 branches but I realized it was still messed up so I fixed everything on this one.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -31,8 +33,4 @@ def full_name
   def admin?
     false
   end
-
-#  def to_param
-#    slug
-#  end
```
Comment (kpearson) [inline app/models/user.rb]: Im hoping to keep this for now so we can talk about it as a group.

File: app/views/application/_host_profile_dropdown.html.erb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 <li class="dropdown">
   <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false" data-no-turbolink>Profile <span class="caret"></span></a>
   <ul class="dropdown-menu" role="menu">
-    <li><%= link_to "View Profile", user_path(current_user) %></li>
+    <li><%= link_to "View Profile", user_path(current_user.id) %></li>
```
Comment (kpearson) [inline app/views/application/_host_profile_dropdown.html.erb]: I think `.id` here is unnecessary.

File: config/environments/production.rb [hunk 3]
```diff
@@ -76,4 +76,12 @@
...
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  config.fog_credentials = {
+    provider: "AWS",
+    aws_access_key_id: ENV["S3_KEY"],
+    aws_secret_access_key: ENV["S3_SECRET"],
+    region: ENV["S3_REGION"]
+  }
+  config.fog_directory = ENV["S3_BUCKET"]
```
Comment (kpearson) [inline config/environments/production.rb]: This doesn't belong here.

File: spec/feature/user/edits_a_listing_spec.rb [hunk 4]
```diff
@@ -14,7 +14,7 @@
     expect(page).to have_content("Edit listing")
   end
 
-  it "and is redirected to the edited listing and sees the edits" do
+  xit "and is redirected to the edited listing and sees the edits" do
     valid_user = create(:user)
```
Comment (kpearson) [inline spec/feature/user/edits_a_listing_spec.rb]: Why is this commented out?

File: app/controllers/user/listings_controller.rb [hunk 5]
```diff
@@ -20,8 +20,8 @@ def create
       params["listing_images"]["images"].each do |i|
         @listing.listing_images.build(image: i)
       end
-    end
-    if @listing.save
+    elsif @listing.save
+      current_user.update_attributes(role: 1)
```
Comment (kpearson) [inline app/controllers/user/listings_controller.rb]: Whats this change doing?

File: app/views/application/_host_profile_dropdown.html.erb (hunk 2 above)
Comment (Lydias303) [inline app/views/application/_host_profile_dropdown.html.erb]: alright i will get rid of it

File: app/controllers/user/listings_controller.rb (hunk 5 above)
Comment (Lydias303) [inline app/controllers/user/listings_controller.rb]: This is where we change a users role to a host role when a listing is created, this was included in my last pul request but someone deleted it. Without it roles dont change.

File: spec/feature/user/edits_a_listing_spec.rb (hunk 4 above)
Comment (Lydias303) [inline spec/feature/user/edits_a_listing_spec.rb]: It was failing on friday we discussed. You and i discussed it and you said to set it as pending.

File: app/models/user.rb [hunk 6]
```diff
@@ -11,8 +11,17 @@ class User < ActiveRecord::Base
   has_many :listings
   has_many :listing_images, :through => :listings
   has_many :orders
+<<<<<<< HEAD
```
Comment (bryceholcomb) [inline app/models/user.rb]: Looks like these are leftover from a merge

File: app/controllers/user/listings_controller.rb (hunk 5 above)
Comment (kpearson) [inline app/controllers/user/listings_controller.rb]: The way this is configured at the moment if the user includes an image the role does not change and is not redirected anywhere.  The if statement which wraps the image build is separate from the reset of the control flow.

What will work is something like
```
def create
  @listing = current_user.listings.new(listing_params)
  if params["listing_images"]
    params["listing_images"]["images"].each do |i|
      @listing.listing_images.build(image: i)
    end
  end
  if @listing.save
    current_user.update_attributes(role: 1)
    redirect_to user_listing_path(current_user.slug, @listing.id)
  else
    render :new
  end
end
```

**Comment by bryceholcomb** [comment]
Everything looks good!

**Pull request merged**